metadata-ingestion/src/datahub/ingestion/source/redshift/query.py

```diff
@@ -443,3 +448,53 @@ def list_copy_commands_sql(
             start_time=start_time.strftime(redshift_datetime_format),
             end_time=end_time.strftime(redshift_datetime_format),
         )
+
+    @staticmethod
+    def get_temp_table_clause(table_name: str) -> List[str]:
+        if table_name.startswith("#"):
+            return [
+                f"{RedshiftQuery.CREATE_TEMPORARY_TABLE_HASH_CLAUSE}{table_name[1:]}"
+            ]
+
+        return [
+            f"{RedshiftQuery.CREATE_TEMP_TABLE_CLAUSE} {table_name}",
+            f"{RedshiftQuery.CREATE_TEMPORARY_TABLE_CLAUSE} {table_name}",
+        ]
+
+    @staticmethod
+    def _like_statement(column_name: str, value: str) -> str:
+        return f"{column_name} ILIKE '{value}%'"
+
+    @staticmethod
+    def temp_table_ddl_query(tables: List[str]) -> str:
+        like_statements: List[str] = []
+        for table_name in tables:
+            like_statements.extend(
+                [
+                    RedshiftQuery._like_statement(
+                        column_name="SYS.query_text", value=temp_clause
+                    )
+                    for temp_clause in RedshiftQuery.get_temp_table_clause(table_name)
+                ]
+            )
+
+        temp_table_condition: str = " OR ".join(like_statements)
+
+        return f"""
+        SELECT  SVL.pid,
+                SVL.xid,
+                SVL.sequence,
+                SVL.text AS query_text
+        FROM    svl_statementtext SVL
```

Comment:
If we use SYS_QUERY_HISTORY we don't need the self-join -> https://docs.aws.amazon.com/redshift/latest/dg/SYS_QUERY_HISTORY.html

Reply:
As discussed I shift the query to SYS_QUERY_HISTORY, here caveat is if query size is greater than 4000 then it wouldn't get processed